Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7176.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Xamarin.Forms.Core;
+using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+
+#if UITEST
+using NUnit.Framework;
+using Xamarin.UITest;
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest.Queries;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 7176, "[Enhancement] Label Autofit Font", PlatformAffected.Android | PlatformAffected.iOS)]
+#if UITEST
+	[Category(UITestCategories.Label)]
+#endif
+	public class Issue7176 : TestContentPage
+	{
+		const string GetFontSize = "getTextSize";
+		const string GetAutoSizeMode = "getAutoSizeTextType";
+		const string AutoFitLabel = "Autofit Label";
+		const string AutoFitModeButton = "Autofit Mode Button";
+
+		protected override void Init()
+		{
+			var autofitLabel = new Label
+			{
+				AutomationId = AutoFitLabel,
+				AutoFitText = AutoFitTextMode.FitToWidth,
+				Text = "Welcome to xamarin forms!!",
+				VerticalOptions = LayoutOptions.FillAndExpand,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				BackgroundColor = Color.LightBlue
+			};
+
+			Content = new StackLayout()
+			{
+				Children =
+				{
+					new ApiLabel(),
```

Comment:
The `ApiLabelRenderer` is not part of this change, but throws an unhandled exception when navigating back from this issue page on iOS

>   at Xamarin.Forms.ControlGallery.iOS.ApiLabelRenderer.OnElementChanged (Xamarin.Forms.Platform.iOS.ElementChangedEventArgs`1[TElement] e) [0x00001] in /Users/jfversluis/Source/Xamarin/Xamarin.Forms/Xamarin.Forms.ControlGallery.iOS/ApiLabelRenderer.cs:16 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x00163] in /Users/jfversluis/Source/Xamarin/Xamarin.Forms/Xamarin.Forms.Platform.iOS/VisualElementRenderer.cs:277 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Dispose (System.Boolean disposing) [0x000bf] in /Users/jfversluis/Source/Xamarin/Xamarin.Forms/Xamarin.Forms.Platform.iOS/VisualElementRenderer.cs:359 
  at Xamarin.Forms.Platform.iOS.ViewRenderer`2[TView,TNativeView].Dispose (System.Boolean disposing) [0x0006b] in /Users/jfversluis/Source/Xamarin/Xamarin.Forms/Xamarin.Forms.Platform.iOS/ViewRenderer.cs:123 
  at Xamarin.Forms.Platform.iOS.LabelRenderer.Dispose (System.Boolean disposing) [0x00001] in /Users/jfversluis/Source/Xamarin/Xamarin.Forms/Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs:143 
  at Foundation.NSObject.Dispose () [0x00000] in | cut for brevity |

Reply:
Done!